homework/src/app.js

```diff
@@ -1,11 +1,157 @@
-{
-  /* Your homework goes here */
-  
-  /* STRINGS */
+ /* STRINGS */
+ let myString = "hello,this,is,a,difficult,to,read,sentence";
+//Add the string to your file and log it.
+console.log(myString);
+
+//Log the length of myString.
+console.log(myString.length);
+
+//The commas make that the sentence is quite hard to read. Find a way to remove the commas from the string and replace them with spaces.
+var newString = myString.replace(/,/g, ' ');
+console.log(newString);
+
+
+/*The string doesn't make any sense. Find a way to reverse this string so that it becomes readable.*/
+
+const str = 'dlroW olleH';
+
+function ReverseString(str) { 
+    return str.split('').reverse().join('') 
+ } 
+ console.log(ReverseString('dlroW olleH')) 
 
   /* ARRAYS */
 
-  /* OPERATORS */
+  let favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+//Add a statement that adds your favorite animal say 'turtle' to the existing array.
+favoriteAnimals.push=('turtle'); // I think new array will include 4 object
+console.log(favoriteAnimals);
+
+//Now add another favorite animal to the array but make sure it is placed after 'blowfish' and before 'capricorn'.
+favoriteAnimals.splice(1, 0, 'meerkat');
+console.log(favoriteAnimals);
+
+//Delete 'giraffe' from the array.
+favoriteAnimals.splice(3, 1); 
+console.log(favoriteAnimals);
+
+ /* OPERATORS */
+
+ // 1. Mathematically, if a == b and b == c, then a == c. Can you explain in your words why this is not true for the above snippet?
+
+var a='0';
+var b= 0;
+var c=[];
+
+
+ console.log( a == b ? 'true':'false');
+ console.log( b == c ? 'true':'false');
+ console.log( a == c ? 'true':'false');
+
+ // a is string and c is array and there value are not equal.
```

Comment:
I don't think this is a complete explanation. `b` is a number and `c` is an array but they are still equal when compared with `==` operator. Could you please try to elaborate this further?

Reply:
a==b // true :  types are not same but their values are equal.
b==c// true :  array is numeric , this array is empty and it value lenght means 0 
a==c//false : a is string c is a array and their value are not equal